pokemongo_bot/cell_workers/evolve_pokemon.py

```diff
@@ -0,0 +1,152 @@
+            return False
+
+        # Evolve all is used - Use Lucky egg only at the first tick
+        if self.bot.tick_count is not 1 or not self.use_lucky_egg:
+            return True
+
+        lucky_egg_count = self.bot.item_inventory_count(Item.ITEM_LUCKY_EGG.value)
+
+        # Make sure the user has a lucky egg and skip if not
+        if lucky_egg_count > 0:
+            logger.log('Using lucky egg ... you have {}'.format(lucky_egg_count))
+            response_dict_lucky_egg = self.bot.use_lucky_egg()
+            if response_dict_lucky_egg:
+                result = response_dict_lucky_egg.get('responses', {}).get('USE_ITEM_XP_BOOST', {}).get('result', 0)
+                if result is 1:  # Request success
+                    logger.log('Successfully used lucky egg... ({} left!)'.format(lucky_egg_count - 1), 'green')
+                    return True
+                else:
+                    logger.log('Failed to use lucky egg!', 'red')
+                    return False
+        else:
+            # Skipping evolve so they aren't wasted
+            logger.log('No lucky eggs... skipping evolve!', 'yellow')
+            return False
+
+    def _get_candy_list(self, inventory_items):
+        candies = {}
+        for item in inventory_items:
+            candy = item.get('inventory_item_data', {}).get('candy', {})
+            family_id = candy.get('family_id', 0)
+            amount = candy.get('candy', 0)
+            if family_id > 0 and amount > 0:
+                family = self.bot.pokemon_list[family_id - 1]['Name'] + " candies"
+                candies[family] = amount
+
+        return candies
+
+    def _sort_and_filter(self, inventory_items):
+        pokemons = []
+        logic_to_function = {
+            'or': lambda pokemon: pokemon["cp"] >= self.evolve_above_cp or pokemon["iv"] >= self.evolve_above_iv,
+            'and': lambda pokemon: pokemon["cp"] >= self.evolve_above_cp and pokemon["iv"] >= self.evolve_above_iv
+        }
+        for item in inventory_items:
+            pokemon = item.get('inventory_item_data', {}).get('pokemon_data', {})
+            pokemon_num = int(pokemon.get('pokemon_id', 0)) - 1
+            next_evol = self.bot.pokemon_list[pokemon_num].get('Next Evolution Requirements', {})
+            pokemon = {
+                'id': pokemon.get('id', 0),
+                'num': pokemon_num,
+                'name': self.bot.pokemon_list[pokemon_num]['Name'],
+                'cp': pokemon.get('cp', 0),
+                'iv': self._compute_iv(pokemon),
+                'candies_family': next_evol.get('Name', ""),
+                'candies_amount': next_evol.get('Amount', 0)
+            }
+            if pokemon["id"] > 0 and pokemon["candies_amount"] > 0 and (logic_to_function[self.cp_iv_logic](pokemon)):
+                pokemons.append(pokemon)
+
+        if self.first_evolve_by == "cp":
+            pokemons.sort(key=lambda x: (x['num'], x["cp"], x["iv"]), reverse=True)
```

Comment:
Looks like it works correctly. 

![image](https://cloud.githubusercontent.com/assets/159076/17385184/520673ac-59e1-11e6-8345-5e3445a38be9.png)